Sources/Purchasing/TestStore/TestStorePurchaseHandler.swift

```diff
@@ -18,64 +18,121 @@ import AppKit
-    /// (e.g. there's already a purchase in progress).
+
     @MainActor
-    func purchase(product: TestStoreProduct, completion: @escaping (TestPurchaseResult) -> Void) throws
+    func purchase(product: TestStoreProduct) async -> TestPurchaseResult
+
     #endif // TEST_STORE
 }
 
 /// The object that handles purchases in the Test Store.
 ///
 /// This class is used to handle purchases when using a Test Store API key.
-class TestStorePurchaseHandler: TestStorePurchaseHandlerType {
+actor TestStorePurchaseHandler: TestStorePurchaseHandlerType {
 
     private let systemInfo: SystemInfo
 
-    @MainActor
-    private var purchaseInProgress: Bool = false
+    private var currentPurchaseTask: Task<TestPurchaseResult, Never>?
+    private var purchaseInProgress: Bool {
+        return self.currentPurchaseTask != nil
+    }
 
     init(systemInfo: SystemInfo) {
         self.systemInfo = systemInfo
     }
 
     #if TEST_STORE
-    @MainActor
-    func purchase(product: TestStoreProduct, completion: @escaping (TestPurchaseResult) -> Void) throws {
+
+    func purchase(product: TestStoreProduct) async -> TestPurchaseResult {
         guard !self.purchaseInProgress else {
-            throw ErrorUtils.operationAlreadyInProgressError()
+            return .failure(ErrorUtils.operationAlreadyInProgressError())
         }
-        self.purchaseInProgress = true
-        let completionWrapper: (TestPurchaseResult) -> Void = { @MainActor [weak self] result in
-            self?.purchaseInProgress = false
-            completion(result)
+
+        let newPurchaseTask = Task<TestPurchaseResult, Never> { [weak self] in
+            guard let self else {
+                return .failure(ErrorUtils.unknownError())
+            }
+
+            return await self.presentPurchaseAlert(for: product)
         }
 
-        #if os(iOS) || os(tvOS) || VISION_OS || targetEnvironment(macCatalyst)
-        self.purchaseWithUIKit(product: product, completion: completionWrapper)
-        #elseif os(watchOS)
-        self.purchaseWithWatchKit(product: product, completion: completionWrapper)
-        #elseif os(macOS)
-        self.purchaseWithAppKit(product: product, completion: completionWrapper)
-        #endif
+        self.currentPurchaseTask = newPurchaseTask
+        let purchaseResult = await newPurchaseTask.value
+        self.currentPurchaseTask = nil
+
+        return purchaseResult
     }
+
     #endif // TEST_STORE
 
+    nonisolated private func completePurchase(
+        product: TestStoreProduct, completion: @escaping (TestPurchaseResult) -> Void
+    ) {
+        Task {
+            let transaction = await self.createStoreTransaction(product: product)
+            completion(.success(transaction))
+        }
+    }
+
+    private func createStoreTransaction(product: TestStoreProduct) async -> StoreTransaction {
+        let transactionId = "test_order_" + UUID().uuidString
+        let storefront = await Storefront.currentStorefront
```

Comment:
I see in the Android SDK the `Storefront` is centralized by `BillingAbstract`'s `getStorefront` method ([here](https://github.com/RevenueCat/purchases-android/blob/3a31fcc0f69c011cff2559415c9bb8b2fc143dfc/purchases/src/main/kotlin/com/revenuecat/purchases/common/BillingAbstract.kt#L124)). In iOS, we have `Storefront.currentStorefront` all over the place, which returns StoreKit 2's storefront (the user's Storefront):
https://github.com/RevenueCat/purchases-ios/blob/d68ec4df92732b58059c677ae25319158b524531/Sources/Purchasing/StoreKitAbstractions/Storefront.swift#L80
For now I think it's OK to use that same Storefront until we need to mock it (see https://github.com/RevenueCat/purchases-android/pull/2554#discussion_r2241987004)